pygame.sprite.spritecollide collided argument typing incompatible with collide_mask

**Environment:**

- **Operating system** (e.g. Windows, Linux(Debian), Linux(Ubuntu), Mac): Windows 10, 22H2
- **Python version** (e.g. 3.7.9, 3.8.5) : 3.11.3
- **SDL version** (e.g. SDL 2.0.12): 2.28.3
- **PyGame version** (e.g. 2.0.0.dev10, 1.9.6): 2.5.2
- **Relevant hardware** (e.g. if reporting a bug about a controller, tell us the brand & name of it): N/A

**Current behavior:**

When `pygame.sprite.collidesprite` or `pygame.sprite.collidespriteany` are called with the `collided` argument set to `pygame.sprite.collide_mask`, mypy is raising an argument type error ([arg-type](https://mypy.readthedocs.io/en/stable/error_code_list.html#check-argument-types-arg-type)).

**Expected behavior:**

The two functions can be used with `collide_mask`. This issue does not happen when setting the `collided` argument to `pygame.sprite.collide_circle` or `pygame.sprite.collide_rect`.

**Steps to reproduce:**

1. Create a sprite and a sprite group
2. Use either the `spritecollide` or `spritecollideany` methods with the `collided` argument set to `pygame.sprite.collide_mask`
3. Run mypy and check output

**Test code**

```py
import pygame

pygame.init()

class ExampleSprite(pygame.sprite.Sprite):
    def __init__(self) -> None:
        super().__init__()

        self.image = pygame.surface.Surface((100, 100)).convert()
        self.rect = self.image.get_rect()

sprite = ExampleSprite()
sprite_group = pygame.sprite.Group(ExampleSprite())

pygame.sprite.spritecollideany(sprite, sprite_group, pygame.sprite.collide_rect) # Ok
pygame.sprite.spritecollideany(sprite, sprite_group, pygame.sprite.collide_circle) # Ok
pygame.sprite.spritecollideany(sprite, sprite_group, pygame.sprite.collide_mask) # Mypy error

pygame.sprite.spritecollide(sprite, sprite_group, False, pygame.sprite.collide_rect) # Ok
pygame.sprite.spritecollide(sprite, sprite_group, False, pygame.sprite.collide_circle) # Ok
pygame.sprite.spritecollide(sprite, sprite_group, False, pygame.sprite.collide_mask) # Mypy error
```

**Stack trace/error output/other error logs**
Mypy output:

```
example.py:17: error: Argument 3 to "spritecollideany" has incompatible type "Callable[[_HasImageAndRect | _HasMaskAndRect, _HasImageAndRect | _HasMaskAndRect], tuple[int, int] | None]"; expected "Callable[[_HasRect, ExampleSprite], bool] | None"  [arg-type]
example.py:21: error: Argument 4 to "spritecollide" has incompatible type "Callable[[_HasImageAndRect | _HasMaskAndRect, _HasImageAndRect | _HasMaskAndRect], tuple[int, int] | None]"; expected "Callable[[_HasRect, ExampleSprite], bool] | None"  [arg-type]
Found 2 errors in 1 file (checked 1 source file)
```

--- a/buildconfig/stubs/pygame/sprite.pyi
+++ b/buildconfig/stubs/pygame/sprite.pyi
@@ -262,20 +262,26 @@ def collide_mask(
     left: _SupportsCollideMask, right: _SupportsCollideMask
 ) -> Optional[Tuple[int, int]]: ...
 def spritecollide(
-    sprite: _HasRect,
-    group: AbstractGroup[_TSprite],
+    sprite: _TSprite,
+    group: AbstractGroup[_TSprite2],
     dokill: bool | Literal[1] | Literal[0],
-    collided: Optional[Callable[[_HasRect, _TSprite], bool]] = None,
-) -> List[_TSprite]: ...
+    collided: Optional[
+        Callable[[_TSprite, _TSprite2], bool | Optional[Tuple[int, int]]]
+    ] = None,
+) -> List[_TSprite2]: ...
 def groupcollide(
     groupa: AbstractGroup[_TSprite],
     groupb: AbstractGroup[_TSprite2],
     dokilla: bool | Literal[1] | Literal[0],
     dokillb: bool | Literal[1] | Literal[0],
-    collided: Optional[Callable[[_TSprite, _TSprite2], bool]] = None,
+    collided: Optional[
+        Callable[[_TSprite, _TSprite2], bool | Optional[Tuple[int, int]]]
+    ] = None,
 ) -> Dict[_TSprite, List[_TSprite2]]: ...
 def spritecollideany(
-    sprite: _HasRect,
-    group: AbstractGroup[_TSprite],
-    collided: Optional[Callable[[_HasRect, _TSprite], bool]] = None,
-) -> Optional[_TSprite]: ...
+    sprite: _TSprite,
+    group: AbstractGroup[_TSprite2],
+    collided: Optional[
+        Callable[[_TSprite, _TSprite2], bool | Optional[Tuple[int, int]]]
+    ] = None,
+) -> Optional[_TSprite2]: ...

